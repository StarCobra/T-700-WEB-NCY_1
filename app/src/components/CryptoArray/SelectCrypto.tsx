import * as React from 'react';
import { useState } from 'react';
import '../../style/cryptoDisplay.scss';
import { Box, TextField, MenuItem } from '@mui/material';

export default function selectCrypto() {
  const [cryptos, setCryptos] = useState<string[]>([])
  console.log(cryptos)
  const handleChangeSelectedCryptos = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value
    setCryptos(typeof value === 'string' ? value.split(',') : value)
  }
  return (
    <div className="selectCryptoContainer">
        <span>Choose your crypto(s) :</span>
        <span>
          <Box className="SelectCryptoBox" color="warning">
            <TextField 
              label="Select a crypto" 
              select 
              className='SelectCryptoLabel' 
              value={cryptos} 
              onChange={handleChangeSelectedCryptos} 
              fullWidth
              SelectProps={{ multiple:true }}
              color='warning'
            >
              <MenuItem value="0">BTC</MenuItem>
              <MenuItem value="1">ETH</MenuItem>
              <MenuItem value="2">BNB</MenuItem>
              <MenuItem value="3">USDC</MenuItem>
              <MenuItem value="4">SOL</MenuItem>
              <MenuItem value="5">ADA</MenuItem>
            </TextField>
          </Box>
        </span>
    </div>
  )
}


const cryptosExample=[{id:0, name:"BTC"}, {id:1, name:"ETH"}, {id:2, name:"BNB"}, {id:3, name:"USDC"}, {id:4, name:"SOL"}, {id:5, name:"ADA"}]