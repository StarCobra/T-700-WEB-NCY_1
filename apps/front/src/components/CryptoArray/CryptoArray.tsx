import * as React from 'react';
import '../../style/cryptoDisplay.scss';
import { TableContainer, Table, TableHead, TableBody, TableRow, TableCell, Paper } from '@mui/material';

export default function CryptoArray() {
  return (
    <div className='cryptoTableContainer'>
        <div className="cryptoArray">
            <TableContainer component={Paper}>
                <Table aria-aria-label='Crypto Array'>
                    <TableHead>
                        <TableRow>
                            <TableCell>Crypto name</TableCell>
                            <TableCell>Last price</TableCell>
                            <TableCell>24h change</TableCell>
                            <TableCell>24h max</TableCell>
                            <TableCell>24h min</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        <TableRow>
                            <TableCell>BTC</TableCell>
                            <TableCell>36 772</TableCell>
                            <TableCell className='negativePercentage'>-1,09</TableCell>
                            <TableCell>37 443</TableCell>
                            <TableCell>36 567</TableCell>
                        </TableRow>
                        <TableRow>
                            <TableCell>ETH</TableCell>
                            <TableCell>2 086</TableCell>
                            <TableCell className='positivePercentage'>1,27</TableCell>
                            <TableCell>2 105</TableCell>
                            <TableCell>2 028</TableCell>
                        </TableRow>
                        <TableRow>
                            <TableCell>Test</TableCell>
                            <TableCell>36 772</TableCell>
                            <TableCell className='negativePercentage'>-1,09</TableCell>
                            <TableCell>37 443</TableCell>
                            <TableCell>36 567</TableCell>
                        </TableRow> 
                        <TableRow>
                            <TableCell>Test2</TableCell>
                            <TableCell>36 772</TableCell>
                            <TableCell className='negativePercentage'>-1,09</TableCell>
                            <TableCell>37 443</TableCell>
                            <TableCell>36 567</TableCell>
                        </TableRow> 
                        <TableRow>
                            <TableCell>Test3</TableCell>
                            <TableCell>36 772</TableCell>
                            <TableCell className='negativePercentage'>-1,09</TableCell>
                            <TableCell>37 443</TableCell>
                            <TableCell>36 567</TableCell>
                        </TableRow> 
                        <TableRow>
                            <TableCell>Test4</TableCell>
                            <TableCell>36 772</TableCell>
                            <TableCell className='negativePercentage'>-1,09</TableCell>
                            <TableCell>37 443</TableCell>
                            <TableCell>36 567</TableCell>
                        </TableRow> 
                        <TableRow>
                            <TableCell>Test5</TableCell>
                            <TableCell>36 772</TableCell>
                            <TableCell className='negativePercentage'>-1,09</TableCell>
                            <TableCell>37 443</TableCell>
                            <TableCell>36 567</TableCell>
                        </TableRow> 
                    </TableBody>
                </Table>
            </TableContainer>
        
        </div>
    </div>
  )
}
 