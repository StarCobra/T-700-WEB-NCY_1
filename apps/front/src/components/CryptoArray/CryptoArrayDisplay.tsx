import React from "react"
import CryptoArray from "./CryptoArray"
import SelectCrypto from "./SelectCrypto"

export default function CryptoArrayDisplay() {
  return (
    <div className="ArrayDisplay">
      <SelectCrypto />

      <CryptoArray />
    </div>
  )
}
