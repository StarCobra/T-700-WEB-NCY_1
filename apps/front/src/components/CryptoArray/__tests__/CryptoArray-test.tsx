import React from "react";
import { act, create } from "react-test-renderer";
import CryptoArray from "../CryptoArray";

describe("Components > CryptoArray > CryptoArray", () => {
  const tests = [
    {
      description: "generic",
      props: {},
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;
      await act(async () => {
        root = create(<CryptoArray {...test.props} />);
      });
      expect(root).toMatchSnapshot();
    });
  });
});
