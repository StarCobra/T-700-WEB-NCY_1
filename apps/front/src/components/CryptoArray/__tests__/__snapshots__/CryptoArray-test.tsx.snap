// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > CryptoArray > CryptoArray should render the component generic 1`] = `
<div
  className="cryptoTableContainer"
>
  <div
    className="cryptoArray"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-rorn0c-MuiTableContainer-root{width:100%;overflow-x:auto;}",
        }
      }
      data-emotion="css rorn0c-MuiTableContainer-root"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-11xur9t-MuiPaper-root-MuiTableContainer-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);width:100%;overflow-x:auto;}",
        }
      }
      data-emotion="css 11xur9t-MuiPaper-root-MuiTableContainer-root"
    />
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-rqglhn-MuiTable-root{display:table;width:100%;border-collapse:collapse;border-spacing:0;}.css-rqglhn-MuiTable-root caption{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;padding:16px;color:rgba(0, 0, 0, 0.6);text-align:left;caption-side:bottom;}",
          }
        }
        data-emotion="css rqglhn-MuiTable-root"
      />
      <table
        aria-aria-label="Crypto Array"
        className="MuiTable-root css-rqglhn-MuiTable-root"
        role={null}
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-15wwp11-MuiTableHead-root{display:table-header-group;}",
            }
          }
          data-emotion="css 15wwp11-MuiTableHead-root"
        />
        <thead
          className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1q1u3t4-MuiTableRow-root{color:inherit;display:table-row;vertical-align:middle;outline:0;}.css-1q1u3t4-MuiTableRow-root.MuiTableRow-hover:hover{background-color:rgba(0, 0, 0, 0.04);}.css-1q1u3t4-MuiTableRow-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-1q1u3t4-MuiTableRow-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}",
              }
            }
            data-emotion="css 1q1u3t4-MuiTableRow-root"
          />
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1ygcj2i-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.5rem;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:left;padding:16px;color:rgba(0, 0, 0, 0.87);}",
                }
              }
              data-emotion="css 1ygcj2i-MuiTableCell-root"
            />
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Crypto name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Last price
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              24h change
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              24h max
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              24h min
            </th>
          </tr>
        </thead>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-apqrd9-MuiTableBody-root{display:table-row-group;}",
            }
          }
          data-emotion="css apqrd9-MuiTableBody-root"
        />
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1ex1afd-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:left;padding:16px;color:rgba(0, 0, 0, 0.87);}",
                }
              }
              data-emotion="css 1ex1afd-MuiTableCell-root"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              BTC
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 772
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium negativePercentage css-1ex1afd-MuiTableCell-root"
            >
              -1,09
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              37 443
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 567
            </td>
          </tr>
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              ETH
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              2 086
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium positivePercentage css-1ex1afd-MuiTableCell-root"
            >
              1,27
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              2 105
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              2 028
            </td>
          </tr>
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Test
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 772
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium negativePercentage css-1ex1afd-MuiTableCell-root"
            >
              -1,09
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              37 443
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 567
            </td>
          </tr>
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Test2
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 772
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium negativePercentage css-1ex1afd-MuiTableCell-root"
            >
              -1,09
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              37 443
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 567
            </td>
          </tr>
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Test3
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 772
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium negativePercentage css-1ex1afd-MuiTableCell-root"
            >
              -1,09
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              37 443
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 567
            </td>
          </tr>
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Test4
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 772
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium negativePercentage css-1ex1afd-MuiTableCell-root"
            >
              -1,09
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              37 443
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 567
            </td>
          </tr>
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Test5
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 772
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium negativePercentage css-1ex1afd-MuiTableCell-root"
            >
              -1,09
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              37 443
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              36 567
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
